### ISSUE TITLE
Consider supporting Docker HEALTHCHECK

### ISSUE THREAD

**Issue by jerrinot** [hazelcast/hazelcast-docker#67]
See https://docs.docker.com/engine/reference/builder/#healthcheck

**Comment by leszko** [comment]
@mesutcelik You can start a container with the health params, but why not to build it into Dockerfile? Then, the user wouldn't need to specify anything.

**Comment by mesutcelik** [comment]
yes sure we can add it but what I am trying to say is that people can already do it if they wanna do it. Even with the existing images... I wasn't sure about the use case too. K8S is already doing it for us.

All being said, adding it easy so lets add then...

**Comment by leszko** [comment]
I've changed my mind here and I think we should not add HEALTHCHECK to our Docker images for the following reasons:
1. The line to add it looks as follows:
```
HEALTHCHECK --interval=1m --timeout=3s CMD curl -f http://localhost:5701/hazelcast/health/node-state || exit 1
```
It means that the port is somehow hardcoded in the health check. So, someone may change the hazelcast port, but forget to change the HEALTHCHECK. It would result in the unhealthy container. Given the fact mentioned in the point 2, I think it may happen quite often. Obviously, we could create a variable PORT, but I'm not sure it will simplify the whole thing for the end user.

2. HEALTHCHECK is not popular, actually I haven't seen any Dockerfile implementing it.

3. HEALTHCHECK is not used by k8 (so the main purpose would be Docker Swarm, which we don't officially support).

4. HEALTHCHECK would use `/health` endpoint which is disabled by default. We would need to enable it by deafult in Docker image.

5. You can still enable it with the command line parameter, so it's not critical to have it.

@jerrinot @mesutcelik , please tell me if you think we should add it anyway. Technically, it's zero effort, I just think we shouldn't do it.

**Comment by mesutcelik** [comment]
Lets just not implement as I commented on https://github.com/hazelcast/hazelcast-docker/issues/67#issuecomment-390307539

Anyone can still use it if they want through `docker run`. Additionally, when I saw there is an option like `HEALTHCHECK NONE` to disable health check inherited from base image, I thought that it has problematic use cases too.
